packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx

```diff
@@ -5,28 +5,28 @@ import utc from 'dayjs/plugin/utc';
 import { Button, OverflowMenu, OverflowMenuItem } from '@carbon/react';
 import { TaskComplete } from '@carbon/react/icons';
 import { useTranslation } from 'react-i18next';
-import { launchOverlay } from '../../hooks/useOverlay';
-import { type MappedAppointment } from '../../types';
+import { closeOverlay, launchOverlay } from '../../hooks/useOverlay';
+import { type Appointment } from '../../types';
 import { showModal } from '@openmrs/esm-framework';
 import { useVisits } from '../../hooks/useVisits';
-import AppointmentForm from '../forms/create-edit-form/appointments-form.component';
+import AppointmentForm from '../../form/appointments-form.component';
 import CheckInButton from './checkin-button.component';
-import DefaulterTracingForm from '../forms/defaulter-tracing-form/default-tracing-form.component';
 
 dayjs.extend(utc);
 dayjs.extend(isToday);
 
 interface AppointmentActionsProps {
   visits: Array<any>;
-  appointment: MappedAppointment;
+  appointment: Appointment;
   scheduleType: string;
+  mutate: () => void;
 }
 
-const AppointmentActions: React.FC<AppointmentActionsProps> = ({ visits, appointment, scheduleType }) => {
+const AppointmentActions: React.FC<AppointmentActionsProps> = ({ visits, appointment, mutate }) => {
```

Comment:
We seem to be passing `mutate()` through three or for different levels, just so we can access it as a prop in `AppointmentForm`. This seems like a perfect place for using a React [context](https://react.dev/reference/react/createContext).

The [PatientSearchContext](https://github.com/openmrs/openmrs-esm-patient-management/blob/90ca8beba97580524d5ccfed10eb885e834589f1/packages/esm-patient-search-app/src/patient-search-context.ts#L2) is a pretty good example of how this can be used.

Reply:
Yeah, it was annoying me that I had to pass this all the way around.  I can check out out PatientSearchContext.

Another pattern I was seeing was using the useSWR call at multiple levels just to retrieve the mutate function, but that didn't seem correct, and seemed dependent on making sure you were asking for the same/correct key at multiple levels.

Reply:
You can always create a new hook that does your `useSWR` call. Then you are sure to be making the same one each time. You shouldn't be passing `mutate`s around. If you don't like the aesthetic of using a fetch hook just to obtain the mutate, then you can also write a custom hook that just obtains the correct mutate function. See https://github.com/openmrs/openmrs-esm-patient-chart/blob/e52c2e74ff9a3924941acf91252dfd892a72e8bf/packages/esm-patient-orders-app/src/api/api.ts#L16 for example. Ignore the comment, it's out of date. https://github.com/openmrs/openmrs-esm-patient-chart/pull/1660